flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -1286,10 +1289,38 @@ private void transferAllDataFromStateHandles(
 			Map<StateHandleID, StreamStateHandle> stateHandleMap,
 			Path restoreInstancePath) throws IOException {
 
+			ExecutorService executorService = Executors.newFixedThreadPool(5);
```

Comment:
Number of threads should be configurable. For example, the same way as incremental checkpointing. At least in RocksDBStateBackend constructor and CheckpointingOptions. The default behaviour (e.g. number of threads 1) can be as it is now, basically running in current thread.